symcrypt-sys/build/static_link.rs

```diff
@@ -0,0 +1,433 @@
+    Ok(())
+}
+
+// TODO: update -symcrypt_fips_build comment
+
+/// Holds configuration options for compiling SymCrypt.
+///
+/// - `triple`: The target triple
+/// - `symcrypt_use_asm`: Whether to enable assembly optimizations.
+/// - `symcrypt_fips_build`: Whether to build in FIPS mode PLACEHOLDER
+/// - `preconfiged_cc`: Returns a Pre-configured `cc` object for the target triple.
+#[derive(Debug)]
+struct SymCryptOptions {
+    triple: Triple,
+    symcrypt_use_asm: bool,
+    //symcrypt_fips_build: bool, // TODO: Determine if we should expose FIPS build option?
+}
+
+impl SymCryptOptions {
+    fn new() -> Self {
+        Self {
+            triple: Triple::get_target_triple(),
+            symcrypt_use_asm: false, // FIXME: Turn this to true when we get ASM checked in
+                                     //symcrypt_fips_build: false, // TODO: Determine if we should expose FIPS build option?
+        }
+    }
+    fn use_asm(&self) -> bool {
+        self.symcrypt_use_asm
+    }
+    fn triple(&self) -> Triple {
+        self.triple.clone()
+    }
+
+    // Returns a cc object that has been preconfigured for the target triple
+    fn preconfigure_cc(&self) -> cc::Build {
+        let mut cc = cc::Build::new();
+        cc.target(self.triple.to_triple())
+            .include("inc")
+            .include("symcrypt/inc")
+            .include("symcrypt/lib")
+            .warnings(false); // Ignore noisy warnings from SymCrypt
+
+        if !self.symcrypt_use_asm {
+            cc.define("SYMCRYPT_IGNORE_PLATFORM", None); // TODO: Fix when we get ASM
+        }
+
+        // Set specific flags for each target
+        match self.triple {
+            Triple::x86_64_pc_windows_msvc => {
+                cc.define("_AMD64_", None);
+                // cc.asm_flag("/DSYMCRYPT_MASM"); // MASM assembly TODO: enable for ASM
+                cc.flag("/MP"); // Multi-threaded compilation
+                cc.flag("/Zp8"); // Structure packing alignment
+                cc.flag("/WX"); // Treat warnings as errors
+                cc.flag("/guard:cf"); // Control Flow Guard
+                                      // cc.flag("/dynamicbase"); // Enable ASLR
+                cc.flag("/EHsc"); // Exception handling
+            }
+            Triple::aarch64_pc_windows_msvc => {
+                cc.define("_ARM64_", None);
+            }
+            Triple::x86_64_unknown_linux_gnu => {
+                cc.flag("-mpclmul");
+                cc.flag("-Wno-incompatible-pointer-types"); // Ignore noisy SymCrypt errors
+                                                            // From SymCrypt-Platforms.cmake
+                cc.flag("-mssse3");
+                cc.flag("-mxsave");
+                cc.flag("-maes");
+                cc.flag("-msha");
+                cc.flag("-mrdrnd");
+                cc.flag("-mrdseed");
+                cc.flag("-mbmi2");
+                cc.flag("-fstack-protector-strong");
+                cc.flag("-Wstack-protector");
```

Comment:
Missing "--param ssp-buffer-size=4". Also why not enable stack protection for all targets?

Reply:
I think stack protection only applies to linux:

https://github.com/microsoft/SymCrypt/blob/53be637dab201a4c9d95e1cf58040c85d71cf3c2/CMakeLists.txt#L79C1-L83C8

```
option(
    SYMCRYPT_STACK_PROTECTION
    "When enabled, SymCrypt will enable various stack protection mechanisms to protect against buffer overruns and the like. Only
    applicable to non-Windows systems. Defaults to ON, but may need to be disabled for specialized targets such as embedded systems."
    ON)
```